buildrunner/config_model.py

```diff
@@ -0,0 +1,154 @@
+                    name = f'{name}:latest'
+
+            if isinstance(push, StepPushDict):
+                names = [f"{push.repository}:{tag}" for tag in push.tags]
+
+            if names is not None:
+                for current_name in names:
+                    if current_name in mp_push_tags:
+                        # raise ValueError(f'Cannot specify duplicate tag {current_name} in build step {step_name}')
+                        raise ValueError(f'Cannot specify duplicate tag {current_name} in build step {step_name}')
+
+            if name is not None and name in mp_push_tags:
+                # raise ValueError(f'Cannot specify duplicate tag {name} in build step {step_name}')
+                raise ValueError(f'Cannot specify duplicate tag {name} in build step {step_name}')
+
+            if update_mp_push_tags and names is not None:
+                mp_push_tags.update(names)
+
+            if update_mp_push_tags and name is not None:
+                mp_push_tags.add(name)
+
+    def validate_multi_platform_build(self, mp_push_tags: Set[str]):
+        """
+        Validate multi-platform build steps
+
+        Args:
+            mp_push_tags (Set[str]): Set of all tags used in multi-platform build steps
+
+        Raises:
+            ValueError | pydantic.ValidationError: If the config file is invalid
+        """
+        # Iterate through each step
+        for step_name, step in self.steps.items():
+            if step.is_multi_platform():
+                if step.build.platform is not None:
+                    raise ValueError(f'Cannot specify both platform ({step.build.platform}) and '
+                                     f'platforms ({step.build.platforms}) in build step {step_name}')
+
+                if not isinstance(step.build.platforms, list):
+                    raise ValueError(f'platforms must be a list in build step {step_name}')
+
+                # Check for valid push section, duplicate mp tags are not allowed
+                self.validate_push(step.push, mp_push_tags, step_name)
+
+    def validate(self):
+        """
+        Validate the config file
+
+        Raises:
+            ValueError | pydantic.ValidationError : If the config file is invalid
+        """
+        if self.has_multi_platform_build():
+            mp_push_tags = set()
+            self.validate_multi_platform_build(mp_push_tags)
+
+            # Validate that all tags are unique across all multi-platform step
+            for step_name, step in self.steps.items():
+
+                # Check that there are no single platform tags that match multi-platform tags
+                if not step.is_multi_platform():
+                    if step.push is not None:
+                        self.validate_push(push=step.push,
+                                           mp_push_tags=mp_push_tags,
+                                           step_name=step_name,
+                                           update_mp_push_tags=False)
```

Comment:
I would add another method to this class or module that constructs the config model, catches ValidationErrors and then makes a more readable error or list of errors out of it: https://git.corp.adobe.com/gauntlet/gauntlet-service-config/blob/master/gauntletconfig/validation/base/valid_yaml.py#L18

Reply:
done